### PR TITLE
Add io.github.alfianlosari.GTKChatGPT

### PR DESCRIPTION
**Pull request by alfianlosari**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly. A native GNOME application that lets you interact with OpenAI's ChatGPT API using your own API Key.
- [X] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:https://github.com/alfianlosari/GTKChatGPT/commits?author=alfianlosari**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.alfianlosari.GTKChatGPT.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            ],
+            "build-commands": [
+                "./setup-offline.sh",
+                "swift build -c release --static-swift-stdlib --skip-update",
+                "install -Dm755 .build/release/GTKChatGPT /app/bin/GTKChatGPT",
+                "install -d /app/bin/GPTEncoder_GPTEncoder.resources",
+                "cp -r .build/release/GPTEncoder_GPTEncoder.resources/* /app/bin/GPTEncoder_GPTEncoder.resources/",
+                "install -Dm644 data/io.github.alfianlosari.GTKChatGPT.metainfo.xml $DESTDIR/app/share/metainfo/io.github.alfianlosari.GTKChatGPT.metainfo.xml",
+                "install -Dm644 data/io.github.alfianlosari.GTKChatGPT.desktop $DESTDIR/app/share/applications/io.github.alfianlosari.GTKChatGPT.desktop",
+                "install -Dm644 data/icons/io.github.alfianlosari.GTKChatGPT.png $DESTDIR/app/share/icons/hicolor/scalable/apps/io.github.alfianlosari.GTKChatGPT.png"
```
Comment (bbhtt) [inline io.github.alfianlosari.GTKChatGPT.json:54]:
```suggestion
                "install -Dm644 data/io.github.alfianlosari.GTKChatGPT.metainfo.xml /app/share/metainfo/io.github.alfianlosari.GTKChatGPT.metainfo.xml",
                "install -Dm644 data/io.github.alfianlosari.GTKChatGPT.desktop /app/share/applications/io.github.alfianlosari.GTKChatGPT.desktop",
                "install -Dm644 data/icons/io.github.alfianlosari.GTKChatGPT.png /app/share/icons/hicolor/scalable/apps/io.github.alfianlosari.GTKChatGPT.png"
```

**Comment by bbhtt** [comment]
bot, build io.github.alfianlosari.GTKChatGPT

File: io.github.alfianlosari.GTKChatGPT.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            ],
+            "build-commands": [
+                "./setup-offline.sh",
+                "swift build -c release --static-swift-stdlib --skip-update",
+                "install -Dm755 .build/release/GTKChatGPT /app/bin/GTKChatGPT",
+                "install -d /app/bin/GPTEncoder_GPTEncoder.resources",
+                "cp -r .build/release/GPTEncoder_GPTEncoder.resources/* /app/bin/GPTEncoder_GPTEncoder.resources/",
+                "install -Dm644 data/io.github.alfianlosari.GTKChatGPT.metainfo.xml /app/share/metainfo/io.github.alfianlosari.GTKChatGPT.metainfo.xml",
+                "install -Dm644 data/io.github.alfianlosari.GTKChatGPT.desktop /app/share/applications/io.github.alfianlosari.GTKChatGPT.desktop",
+                "install -Dm644 data/icons/io.github.alfianlosari.GTKChatGPT.png /app/share/icons/hicolor/scalable/apps/io.github.alfianlosari.GTKChatGPT.png"
```
Comment (bbhtt) [inline io.github.alfianlosari.GTKChatGPT.json:54]: pngs go to 128x128, 256x256 etc. subfolders, scalable is for scalable icons.

Comment (alfianlosari) [reply io.github.alfianlosari.GTKChatGPT.json]: Updated

**Comment by bbhtt** [comment]
bot, build io.github.alfianlosari.GTKChatGPT

**Comment by bbhtt** [comment]
is it intentional that the repo says GTKchatgpt while the name says XCA AI chat?

File: io.github.alfianlosari.GTKChatGPT.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                },
+                "generated-sources.json"
+            ],
+            "build-commands": [
+                "./setup-offline.sh",
+                "swift build -c release --static-swift-stdlib --skip-update",
+                "install -Dm755 .build/release/GTKChatGPT /app/bin/GTKChatGPT",
+                "install -d /app/bin/GPTEncoder_GPTEncoder.resources",
+                "cp -r .build/release/GPTEncoder_GPTEncoder.resources/* /app/bin/GPTEncoder_GPTEncoder.resources/",
+                "install -Dm644 data/io.github.alfianlosari.GTKChatGPT.metainfo.xml /app/share/metainfo/io.github.alfianlosari.GTKChatGPT.metainfo.xml",
```
Comment (bbhtt) [inline io.github.alfianlosari.GTKChatGPT.json:52]: you forgot to fill up the developer name `org.example`...

https://github.com/alfianlosari/GTKChatGPT/blob/68d067f647a0acc8d179d38347a0b8852d0dba87/data/io.github.alfianlosari.GTKChatGPT.metainfo.xml#L6-L8

Comment (bbhtt) [reply io.github.alfianlosari.GTKChatGPT.json:52]: `xml:lang="en"` is not needed https://github.com/alfianlosari/GTKChatGPT/blob/68d067f647a0acc8d179d38347a0b8852d0dba87/data/io.github.alfianlosari.GTKChatGPT.metainfo.xml#L21-L27

Comment (bbhtt) [reply io.github.alfianlosari.GTKChatGPT.json:52]: drop `en` translation https://github.com/alfianlosari/GTKChatGPT/blob/68d067f647a0acc8d179d38347a0b8852d0dba87/data/io.github.alfianlosari.GTKChatGPT.metainfo.xml#L10 the source is already `en` ...

**Comment by alfianlosari** [comment]
> is it intentional that the repo says GTKchatgpt while the name says XCA AI chat?

Yes, it is.

File: io.github.alfianlosari.GTKChatGPT.json (hunk 3 above)
Comment (alfianlosari) [reply io.github.alfianlosari.GTKChatGPT.json:52]: Updated

**Comment by bbhtt** [comment]
bot, build io.github.alfianlosari.GTKChatGPT

**Comment by bbhtt** [comment]
I generated an API key using a test account but it seems to always say

```
Error:
Status Code: 500. Check OpenAI Doc for status code error description.

Something went wrong. Please check your API key, model access, or try again later.
```

whenever I send a message. Is this working?

**Comment by alfianlosari** [comment]
> I generated an API key using a test account but it seems to always say
> 
> ```
> Error:
> Status Code: 500. Check OpenAI Doc for status code error description.
> 
> Something went wrong. Please check your API key, model access, or try again later.
> ```
> 
> whenever I send a message. Is this working?

I suspect the issue might be related to billing. I think as of now, OpenAI doesn’t provide any free credit/plan when using an API key. Could you confirm if the account you use has valid payment method (e.g credit card for billing or purchasing prepaid credits)

IMG_5725

I’m using it with my account which has prepaid credit and it works fine. I’m on Tier 2 which has access to gpt-4o models.

**Comment by bbhtt** [comment]
It's a test account, so no credit is attached to it. Probably better to add the info to a help box inside settings.

**Comment by alfianlosari** [comment]
> It's a test account, so no credit is attached to it. Probably better to add the info to a help box inside settings.

Thanks for the suggestion, will add it in the next release 🙏🏻

**Pull request closed without merging**